library-api/src/main/java/au/csiro/pathling/library/PathlingContextConfiguration.java

```diff
@@ -131,24 +131,34 @@ TerminologyAuthConfiguration toAuthConfig() {
 
 
   @Nonnull
-  HttpClientConf toClientConfig() {
-    final HttpClientConf c = HttpClientConf.defaults();
-    c.setMaxConnectionsTotal(
-        DEFAULT_MAX_TOTAL_CONNECTIONS.resolve(getMaxConnectionsTotal()));
-    c.setMaxConnectionsPerRoute(
+  HttpClientConfiguration toClientConfig() {
+    final HttpClientConfiguration config = HttpClientConfiguration.defaults();
+    config.setMaxConnectionsTotal(DEFAULT_MAX_TOTAL_CONNECTIONS.resolve(getMaxConnectionsTotal()));
+    config.setMaxConnectionsPerRoute(
         DEFAULT_MAX_CONNECTIONS_PER_ROUTE.resolve(getMaxConnectionsPerRoute()));
-    c.setSocketTimeout(DEFAULT_SOCKET_TIMEOUT.resolve(getTerminologySocketTimeout()));
-    return c;
+    config.setSocketTimeout(DEFAULT_SOCKET_TIMEOUT.resolve(getTerminologySocketTimeout()));
+    return config;
   }
 
   @Nonnull
-  HttpCacheConf toCacheConfig() {
-    final HttpCacheConf c = HttpCacheConf.defaults();
-    c.setEnabled(nonNull(getCacheStorageType()));
-    c.setStorageType(DEFAULT_CACHE_STORAGE_TYPE.resolve(getCacheStorageType()));
-    c.setMaxCacheEntries(DEFAULT_CACHE_MAX_ENTRIES.resolve(getCacheMaxEntries()));
-    c.setMaxObjectSize(DEFAULT_CACHE_MAX_OBJECT_SIZE.resolve(getCacheMaxObjectSize()));
-    c.setStorage(getCacheStorageProperties());
-    return c;
+  HttpClientCachingConfiguration toCacheConfig() {
+    final HttpClientCachingConfiguration config = HttpClientCachingConfiguration.defaults();
+    final boolean enabled = nonNull(getCacheStorageType());
+    config.setEnabled(enabled);
+    if (enabled) {
+      config.setMaxEntries(DEFAULT_CACHE_MAX_ENTRIES.resolve(getCacheMaxEntries()));
+      config.setMaxObjectSize(DEFAULT_CACHE_MAX_OBJECT_SIZE.resolve(getCacheMaxObjectSize()));
+      final StorageType storageType = StorageType.fromCode(getCacheStorageType());
+      if (storageType == null) {
+        throw new IllegalArgumentException("Invalid cache storage type: " + getCacheStorageType());
+      }
```

Comment:
Maybe move the exception to StorageType.fromCode()
```
final StorageType storageType = StorageType.fromCode(requireNonNull(getCacheStorageType()));
```

Reply:
Would this handle the case where someone passes in a nonsense storage code through the library API? Would the error back to the user be descriptive enough for them to correct it?